src/io/sudhir/switchboard/Supply.java

```diff
@@ -1,9 +1,9 @@
 package io.sudhir.switchboard;
 
+import com.google.common.collect.ImmutableList;
 import java.util.Optional;
-import java.util.stream.Stream;
 
 public interface Supply {
 
-  Optional<Choice> estimateFor(Demand demand, Stream<Choice> commitments);
+  Optional<Choice> estimateFor(Demand demand, ImmutableList<Choice> commitments);
```

Comment:
Switch this to List - you send an immutable list, but tell external parties that you're sending a List.

Reply:
Don't force all consumers to use the Immutable library.

Reply:
i.e. make your interfaces as generic as possible, but make the actual implementation as safe as possible.

Reply:
You generally want to document the fact that the object is immutable, though. So users don't attempt to modify it. But let users use it as a List or copy it or use any other immutable library, don't force them to use this one.

Reply:
Got it @sudhirj !
Done.